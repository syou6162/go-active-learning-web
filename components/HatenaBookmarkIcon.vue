<template>
  <img 
    v-b-popover.hover.bottomright="tags_and_comment" 
    v-lazy="'https://cdn.profile-image.st-hatena.com/users/' + bookmark.user+ '/profile.png'" 
    :alt="user_id"
    :title="user_id"
    class="hatena-bookmark-user-icon-img"
    target="_blank" rel="noopener"
  >
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import Bookmark from '~/models/Bookmark'

@Component
export default class HatenaBookmarkIcon extends Vue {
  get user_id(): string {
    return 'id:' + this.bookmark.user;
  }
  get tags_and_comment(): string {
    const tags = this.bookmark.tags.map(t => '[' + t + ']').join("")
    return tags + ' '+ this.bookmark.comment;
  }
  @Prop()
  bookmark!: Bookmark;
}
</script>

<style scoped>
.hatena-bookmark-user-icon-img {
  width: 24px;
  height: 24px;
  margin: 2px;
}
</style>