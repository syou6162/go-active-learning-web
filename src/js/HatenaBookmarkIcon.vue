<template>
  <img 
    v-b-popover.hover="'id:' + bookmark.user" 
    v-bind:alt="bookmark.user" 
    style="width: 24px; height: 24px; margin: 2px" 
    v-lazy="'https://cdn.profile-image.st-hatena.com/users/' + bookmark.user+ '/profile.png'" 
    />
</template>

<script>
export default {
  props: ['bookmark'],
}
</script>
