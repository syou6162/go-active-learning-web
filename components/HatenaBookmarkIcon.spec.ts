import { createLocalVue, mount } from '@vue/test-utils'

import HatenaBookmarkIcon from './HatenaBookmarkIcon.vue'
import VueLazyLoad from 'vue-lazyload'
import { VBPopover } from 'bootstrap-vue'

const localVue = createLocalVue();
localVue.use(VueLazyLoad);
localVue.directive('b-popover', VBPopover)

describe('HatenaBookmarkIcon', () => {
  test('はてブのアイコンが正しく表示される', async () => {
    const wrapper = mount(HatenaBookmarkIcon, {
      localVue,
      propsData: {
        bookmark: {
          user: "syou6162",
          tags: ["tag1", "tag2"]
        }
      },
    })
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.props().bookmark.user).toBe("syou6162")
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.attributes("alt")).toBe("id:syou6162")
    expect(wrapper.attributes("data-src")).toBeUndefined()

    wrapper.trigger('mouseover')
    await wrapper.vm.$nextTick()
    expect(wrapper.attributes("data-src")).toBe("https://cdn.profile-image.st-hatena.com/users/syou6162/profile.png")
    expect(wrapper.element).toMatchSnapshot()
  })
})