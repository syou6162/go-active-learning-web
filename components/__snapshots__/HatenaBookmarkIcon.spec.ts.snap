// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HatenaBookmarkIcon はてブのアイコンが正しく表示される 1`] = `
<img
  alt="id:syou6162"
  class="hatena-bookmark-user-icon-img"
  rel="noopener"
  target="_blank"
  title="id:syou6162"
/>
`;

exports[`HatenaBookmarkIcon はてブのアイコンが正しく表示される 2`] = `
<img
  alt="id:syou6162"
  class="hatena-bookmark-user-icon-img"
  data-src="https://cdn.profile-image.st-hatena.com/users/syou6162/profile.png"
  lazy="loading"
  rel="noopener"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  target="_blank"
  title="id:syou6162"
/>
`;
